### PR TITLE
Make integration tests schema based - Closes #150

### LINKED ISSUE #150: Make integration tests schema-based
### Actual behavior

Some integration tests are tested against particular datasets, which change over time on testnet.

```
Expected: ObjectContaining {"address": "5201600508578320196L", "missedBlocks": 194, "productivity": "99.85", "publicKey": "473c354cdf627b82e9113e02a337486dd3afc5615eb71ffd311c5a0beda37b8c", "rank": 1, "secondPublicKey": "02bb04b8b15f10edcd5fbc067c6107841b527a39d57dd33156de616714863bae", "username": "cc001"}
    Received: {"address": "5201600508578320196L", "approval": "88.63", "missedBlocks": 194, "producedBlocks": 133750, "productivity": "99.86", "publicKey": "473c354cdf627b82e9113e02a337486dd3afc5615eb71ffd311c5a0beda37b8c", "rank": 1, "rewards": "46483900000000", "secondPublicKey": "02bb04b8b15f10edcd5fbc067c6107841b527a39d57dd33156de616714863bae", "username": "cc001", "vote": "12586140872901152"}
```

### Expected behavior

Make these tests independent from actual data, testing it against schema.

### Steps to reproduce

Run tests on Jenkins, some will fail. (`/delegates` endpoint)

### PR DESCRIPTION
**Pull request by sameersubudhi**
### What was the problem?
This PR resolves #150 

### How was it solved?
- [x] Make all the integration tests schema-based, agnostic of the environment
- [x] Add additional test cases for missing scenarios and better code coverage
- [x] Refactor all the tests to reuse a single set of the helpers and schema

### How was it tested?
Local + Jenkins

### CODE REVIEW

File: tests/helpers/socketIoRpcRequest.js [hunk 1]
```diff
@@ -15,29 +15,31 @@
...
 
-	socket.emit('request', { method, params }, answer => {
+	socket.emit('request', { jsonrpc: '2.0', method, params }, answer => {
 		socket.close();
-		if (answer.error) {
-			reject(answer);
-		} else {
-			resolve(answer);
-		}
+		resolve(answer);
```
Comment (nagdahimanshu) [inline tests/helpers/socketIoRpcRequest.js:27]:
```suggestion
		if (answer.error) reject(answer);
		resolve(answer);
```

File: tests/integration/api/http/networkSearch.test.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		expect(response).toMap(goodRequestSchema);
+		expect(response.data).toBeArrayOfSize(1);
+		response.data.forEach(delegate => expect(delegate)
+			.toMap(searchItemSchema, { id: q, type: 'tx' }));
+		expect(response.meta).toMap(metaSchema);
+	});
+
+	it('returns a proper error when called without q param', async () => {
+		const response = await api.get(endpoint, 400);
+		expect(response).toMap(notFoundSchema);
```
Comment (nagdahimanshu) [inline tests/integration/api/http/networkSearch.test.js:118]:
```suggestion
		expect(response).toMap(badRequestSchema);
```

File: tests/integration/api/http/networkSearch.test.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	});
+
+	it('returns a proper error when called with empty q param', async () => {
+		const response = await api.get(`${endpoint}?q=`, 400);
+		expect(response).toMap(notFoundSchema);
```
Comment (nagdahimanshu) [inline tests/integration/api/http/networkSearch.test.js:123]:
```suggestion
		expect(response).toMap(badRequestSchema);
```

File: tests/integration/api/http/networkSearch.test.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const config = require('../../../config');
+const { api } = require('../../../helpers/api');
+
+const {
+	notFoundSchema,
```
Comment (nagdahimanshu) [inline tests/integration/api/http/networkSearch.test.js:20]:
```suggestion
	badRequestSchema,
```

**Review by nagdahimanshu — CHANGES_REQUESTED** [review]
looks good apart from small suggestions

**Review by nagdahimanshu — APPROVED** [review]

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**